### DISCUSSION TITLE
How to get single-instance to work?

### DISCUSSION THREAD

**Discussion by arthsmn** [kovidgoyal/kitty#6937] [Q&A] [answered]
When I try to set the command `kitty --single-instance` on KDE Plasma 5 it opens just the first window, if I try to open another one it just doesn't do anything. Same behavior when copying the .desktop to ~/.local/share/applications and changing the Exec option.

**Comment by kovidgoyal** [comment]
On Thu, Dec 21, 2023 at 01:51:30AM -0800, Arthur wrote:
> When I try to set the command `kitty --single-instance` on KDE Plasma 5 it opens just the first window, if I try to open another one it just doesn't do anything. Same behavior when copying the .desktop to ~/.local/share/applications and changing the Exec option.

Works fine for me on KDE Plasma 5 with either wayland or x11.

**Reply by arthsmn** [reply]
Hey, found something. I launched a normal kitty window, then launched `kitty -1`, but when tried to launch `kitty -1` inside the later I got this error in the normal window: `[355 13:43:54.616465] Reading from peer failed: Ignoring too large message from peer`.

**Comment by kovidgoyal** [comment]
what version of kitty are you running?

On Thu, Dec 21, 2023 at 08:45:44AM -0800, Arthur wrote:
> Hey, found something. I launched a normal kitty window, then launched `kitty -1`, but when tried to launch `kitty -1` inside the later I got this error in the normal window: `[355 13:43:54.616465] Reading from peer failed: Ignoring too large message from peer`
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/6937#discussioncomment-7920353
> You are receiving this because you commented.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by arthsmn** [reply]
0.31.0, from nixpkgs. Using NixOS btw.

**Comment by kovidgoyal** [comment, marked as answer]
that will be the issue. Use the official kitty binaries and you will be
fine.

**Comment by Luflosi** [comment]
You should report this issue to Nixpkgs.

**Reply by arthsmn** [reply]
I will, thanks!

**Comment by arthsmn** [comment]
Just as an update, made [an issue](https://github.com/NixOS/nixpkgs/issues/276031) and noticed that when I open from the menu it works fine, maybe it is something with the KDE shortcut system?

**Reply by kovidgoyal** [reply]
That will likley come from having multiple kitty's on your system and different ones being run from different places.
